src/magellan_vision/scripts/yolo.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""
+Run a YOLO_v3 style detection model on test images.
+Adapted from https://github.com/qqwweee/keras-yolo3
+"""
+
+# import colorsys
+import os
+from timeit import default_timer as timer
+
+import numpy as np
+from keras import backend as K
+from keras.models import load_model
+from keras.layers import Input
+from PIL import Image, ImageFont, ImageDraw
+
+from yolo3.model import yolo_eval, yolo_body, tiny_yolo_body
+from yolo3.utils import letterbox_image
+import os
+from keras.utils import multi_gpu_model
+
+class YOLO(object):
+    def __init__(self):
+        self.model_path = 'weights.h5' # model path or trained weights path
+        self.anchors_path = 'yolo_anchors.txt'
+        self.classes_path = 'classes.txt'
+        self.score = 0.1
+        self.iou = 0.45
+        self.class_names = self._get_class()
+        self.anchors = self._get_anchors()
+        self.sess = K.get_session()
+        self.model_image_size = (320, 320) # fixed size or (None, None), hw, multiple of 32
+        self.boxes, self.scores, self.classes = self.generate()
+
+    def _get_class(self):
+        classes_path = os.path.expanduser(self.classes_path)
+        with open(classes_path) as f:
+            class_names = f.readlines()
+        class_names = [c.strip() for c in class_names]
+        return class_names
+
+    def _get_anchors(self):
+        anchors_path = os.path.expanduser(self.anchors_path)
+        with open(anchors_path) as f:
+            anchors = f.readline()
+        anchors = [float(x) for x in anchors.split(',')]
+        return np.array(anchors).reshape(-1, 2)
+
+    def generate(self):
+        model_path = os.path.expanduser(self.model_path)
+        assert model_path.endswith('.h5'), 'Keras model or weights must be a .h5 file.'
+
+        # Load model, or construct model and load weights.
+        num_anchors = len(self.anchors)
+        num_classes = len(self.class_names)
+        is_tiny_version = num_anchors==6 # default setting
+        try:
+            self.yolo_model = load_model(model_path, compile=False)
+        except:
+            self.yolo_model = tiny_yolo_body(Input(shape=(None,None,3)), num_anchors//2, num_classes) \
+                if is_tiny_version else yolo_body(Input(shape=(None,None,3)), num_anchors//3, num_classes)
+            self.yolo_model.load_weights(self.model_path) # make sure model, anchors and classes match
+        else:
+            assert self.yolo_model.layers[-1].output_shape[-1] == \
+                num_anchors/len(self.yolo_model.output) * (num_classes + 5), \
+                'Mismatch between model and given anchor and class sizes'
+
+        print('{} model, anchors, and classes loaded.'.format(model_path))
+
+        # Generate colors for drawing bounding boxes.
+        hsv_tuples = [(x / len(self.class_names), 1., 1.)
+                      for x in range(len(self.class_names))]
+        self.colors = list(map(lambda x: colorsys.hsv_to_rgb(*x), hsv_tuples))
```

Comment:
It might be nice to standardize on list comprehension. I think `[colorsys.hsv_to_rgb(*x) for x in hsv_tuples]` is equivalent?

https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html